lib/fenrir_view/property_types.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module FenrirView
+  class PropertyTypes
+    def initialize(component_class:, component_properties:, instance_properties:)
+      @component_class = component_class
+      @component_properties = component_properties
+      @instance_properties = instance_properties
+    end
+
+    ALL_TYPES = [:default, :required, :one_of_type, :one_of, :note]
+    VALIDATION_TYPES = [:required, :one_of_type, :one_of]
+
+    def validate_properties
+      return if @instance_properties == false # Rendering of this instance has been blocked by a guard clause
+      raise invalid_component_properties unless @instance_properties.is_a?(Hash)
+      raise unknown_keys if instance_has_unknown_keys?
+
+      @instance_properties.each do |property, value|
+        property_validations = @component_properties[property].except(:default, :note)
```

Comment:
This should be `.slice(*VALIDATION_TYPES)`?

Reply:
Ah yes, forgot to update this will do 👍